plutus-tx-plugin/testlib/IntegerLiteral/NoStrict.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE BangPatterns        #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+
+{-# LANGUAGE NoStrict            #-}
+
+-- | We use this module to ensure that integer literals are handled correctly by all
+-- supported GHC versions, when @Strict@ is off (we get different Core depending on
+-- whether or not @Strict@ is on).
+module IntegerLiteral.NoStrict where
+
+import PlutusTx.Code
+import PlutusTx.Prelude qualified as PlutusTx
+import PlutusTx.TH (compile)
+
+integerLiterals :: CompiledCode (Integer -> Integer)
+integerLiterals =
+  $$( compile
+        [||
+        \x ->
+          let !smallStrict = 123
```

Comment:
I remember how I've handled this in the past: `LANGUAGE NegativeLiterals`. That avoids the `integerNegate` calls. Nonetheless, I think handling `integerNegate` is reasonable.

Reply:
`NegativeLiterals` work for most cases (it turns `integerNegate (IP ...)` into `IN ...`), but for double negation, we still get `integerNegate`.

Reply:
You just [told me](https://github.com/input-output-hk/plutus/pull/5494#discussion_r1305671039) the compiler cancels double negation!

Anyway, if we just turn `integerNegate` into a function, it doesn't matter any more.